cmd/client/status.go

```diff
@@ -0,0 +1,119 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/monstercat/gologx/logxhost"
+	"github.com/olekukonko/tablewriter"
+)
+
+func showStatus(name string, args []string) error {
+	var postgres string
+	var services string
+
+	set := flag.NewFlagSet(name, flag.ExitOnError)
+	set.StringVar(&postgres, "postgres", "", "Postgres database")
+	set.StringVar(&services, "services", "", "List of services")
+
+	if err := set.Parse(args); err != nil {
+		return err
+	}
+
+	db, err := logxhost.GetPostgresConnection(postgres)
+	if err != nil {
+		return err
+	}
+	defer db.Close()
+
+	srvcsArr := func() []string {
+		if services != "" {
+			return strings.Split(strings.ReplaceAll(services, " ", ""), ",")
+		} else {
+			return make([]string, 0)
+		}
+	}()
+
+	srvcs, err := logxhost.SelectServices(db, srvcsArr)
+	if err != nil {
+		return err
+	}
+
+	if len(srvcs) == 0 {
+		fmt.Println(styleWarning("No services were found"))
+		os.Exit(1)
+	}
+
+	table := tablewriter.NewWriter(os.Stdout)
+	table.SetHeader([]string{"Id", "Machine", "Status", "Name", "Last Seen"})
+
+	for _, s := range srvcs {
+		row := []string{s.Id, s.Machine, getStatus(s.LastSeen), s.Name, s.LastSeen.Format("2006-01-02 15:04:05")}
+		table.Append(row)
+	}
+
+	table.Render()
+
+	return nil
+}
+
+func getStatus(date time.Time) string {
+
+	year, month, day, hour, min, _ := timeDiff(time.Now(), date)
+	//If heartbeat is bigger than 1 minute, it is inactive.
+	if year == 0 && month == 0 && day == 0 && hour == 0 && min == 0 {
```

Comment:
There is a chance that the current time is 3:00:01 and the heartbeat happened 2:59:59 and it will say inactive.

Reply:
Also, you can just use https://golang.org/pkg/time/#Time.Sub and compare to 60 * time.Second